### ISSUE TITLE
RN 0.76 Regression - RCTWindowFrameDidChangeNotification is not called when dark mode is toggled

### ISSUE THREAD

**Issue by jpudysz** [facebook/react-native#47234]
### Description

I’ve noticed another discrepancy between RN 0.75.x and RN 0.76. With the new bare starter, my iOS native notification listener is no longer called when the user changes the color scheme. Previously, I was using the `RCTWindowFrameDidChangeNotification` notification to receive information about screen and native color scheme changes.

RN 0.75 repo - https://github.com/jpudysz/rn-0.75-jsi-reproducer/tree/feature/repro2

### Steps to reproduce

1. Open both the 0.75 and 0.76 repos
2. Navigate to `Pods/Development Pods/Unistyles/ios/NativePlatformListener+ios.swift`
3. Add a breakpoint at line 29

For RN 0.75 - the event is reported
For RN 0.76 - the breakpoint is not hit

### React Native Version

0.76.0

### Affected Platforms

Runtime - iOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 15.0
  CPU: (12) arm64 Apple M3 Pro
  Memory: 728.19 MB / 36.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.15.0
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.22
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.7.0
    path: /usr/local/bin/npm
  Watchman:
    version: 2024.06.24.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.0
      - iOS 18.0
      - macOS 15.0
      - tvOS 18.0
      - visionOS 2.0
      - watchOS 11.0
  Android SDK:
    API Levels:
      - "23"
      - "28"
      - "29"
      - "30"
      - "33"
      - "34"
      - "35"
    Build Tools:
      - 30.0.3
      - 34.0.0
      - 35.0.0
    System Images:
      - android-23 | ARM 64 v8a
      - android-28 | ARM 64 v8a
      - android-29 | ARM 64 v8a
      - android-30 | ARM 64 v8a
      - android-30 | Google APIs ARM 64 v8a
      - android-34 | Android TV ARM 64 v8a
      - android-34 | ARM 64 v8a
      - android-35 | Google APIs ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2024.2 AI-242.23339.11.2421.12483815
  Xcode:
    version: 16.0/16A242d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.11
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.0.0-alpha.2
    wanted: 15.0.0-alpha.2
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.76.0
    wanted: 0.76.0
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
No crash
```

### Reproducer

https://github.com/jpudysz/rn-0.76-jsi-reproducer/tree/feature/repro2

### Screenshots and Videos

RN 0.75:

Unistyles theme event was applied.

https://github.com/user-attachments/assets/9b70c54c-a971-4b63-a68f-113809583cc5

RN 0.76:

Unistyles theme is not applied as event was not called.

https://github.com/user-attachments/assets/2f093ab3-b94e-4a4f-9779-3964aee8df02

**Comment by jpudysz** [comment]
One more note: both repros are running New Arch, so it's not about it

**Comment by zhongwuzw** [comment]
Issue Fixed in #47236.
